src/process.cpp

```diff
@@ -698,7 +714,11 @@ namespace proc {
     auto proc_opt = proc::parse(file_name);
 
     if (proc_opt) {
-      proc = std::move(*proc_opt);
+      // Update the process object with the new environment and apps
+      // And, keep app running status.
+
+      proc.set_env(proc_opt->get_env());
```

Comment:
After looking into this I have found another bug related to `_env`. It is used in `terminate` function, this means we have to preserve the `_env` until process terminates :/

The moment we call `execute` we should copy the `_env` to a member variable `_app_env` (name is debatable) and replace every usage of `_env` within `execute` and `terminate` with `_app_env` (except the first one where we copy it of course). Or we could store the new ENV to `_current_env` and the in execute copy it (`_env = _current_env`). I prefer the first renaming option, but whatever is easier is ok.